swap/src/common/mod.rs

```diff
@@ -0,0 +1,252 @@
+use uuid::Uuid;
+
+use crate::fs::system_data_dir;
+
+const LATEST_RELEASE_URL: &str = "https://github.com/comit-network/xmr-btc-swap/releases/latest";
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Version {
+    Current,
+    Available,
+}
+
+/// Check the latest release from GitHub API.
+pub async fn check_latest_version(current_version: &str) -> anyhow::Result<Version> {
+    let response = reqwest::get(LATEST_RELEASE_URL).await?;
+    let e = "Failed to get latest release.";
+    let download_url = response.url();
+    let segments = download_url.path_segments().ok_or_else(|| anyhow!(e))?;
+    let latest_version = segments.last().ok_or_else(|| anyhow!(e))?;
+
+    let result = if is_latest_version(current_version, latest_version) {
+        Version::Current
+    } else {
+        tracing::warn!(%current_version, %latest_version, %download_url,
+            "You are not on the latest version",
+        );
+        Version::Available
+    };
+
+    Ok(result)
+}
+
+// todo: naive implementation can be improved using semver
+fn is_latest_version(current: &str, latest: &str) -> bool {
+    current == latest
+}
+
+/// helper macro for [`redact`]... eldrich sorcery
+/// the macro does in essence the following:
+/// 1. create a static regex automaton for the pattern
+/// 2. find all matching patterns using regex
+/// 3. create a placeholder for each distinct matching pattern
+/// 4. add the placeholder to the hashmap
+macro_rules! regex_find_placeholders {
+    ($pattern:expr, $create_placeholder:expr, $replacements:expr, $input:expr) => {{
+        // compile the regex pattern
+        static REGEX: once_cell::sync::Lazy<regex::Regex> = once_cell::sync::Lazy::new(|| {
+            regex::Regex::new($pattern).expect("invalid regex pattern")
+        });
+
+        // keep count of count patterns to generate distinct placeholders
+        let mut counter: usize = 0;
+
+        // for every matched address check whether we already found it
+        // and if we didn't, generate a placeholder for it
+        for address in REGEX.find_iter($input) {
+            if !$replacements.contains_key(address.as_str()) {
+                $replacements.insert(address.as_str().to_owned(), $create_placeholder(counter));
+                counter += 1;
+            }
+        }
+    }};
+}
+
+/// Print the logs from the specified logs or from the default location 
+/// to the specified path or the terminal.
+/// 
+/// If specified, filter by swap id or redact addresses.
+pub async fn print_or_write_logs(logs_dir: Option<PathBuf>, output_path: Option<PathBuf>, swap_id: Option<Uuid>, redact_addresses: bool) -> anyhow::Result<()> {
+    // use provided directory of default one
+    let default_dir = system_data_dir()?.join("logs");
+    let logs_dir = logs_dir.unwrap_or(default_dir);
+
+    tracing::info!("Reading `*.log` files from `{}`", logs_dir.display());
```

Comment:
I think we can remove this. Most of the time we'll be reading from a single log file.